rfd/XXXX-securing-amis.md

```diff
@@ -0,0 +1,225 @@
+---
+authors: Trent Clarke (trent@goteleport.com)
+state: Draft
+---
+
+# RFD 0XXX - Delivering more secure AMIs
+
+# Required Approvers
+* Engineering: @r0mant
+* Security: (@reedloden && @jentfoo)
+* Product: (@xinding33 || @klizhentas)
+
+## What
+
+This RFD discusses structures and processes to increase the security of the
+Amazon EC2 Disk Images (a.k.a AMIs) used to ship Teleport to customers.
+
+## Why
+
+One way our customers consume Teleport is by using a pre-built and  -configured
+AMI to run Teleport in AWS EC2 instances. As a provider of these images we are 
+(at least partially) responsible for the secutity of all ssoftware included in
+the image, not just teleport. 
+
+We should endeavour to not ship vulnerabilities to our clients, even if those
+vulnerabilities are not directly in our software.
+
+This is not simply an academic exercise. Customers have been asking for hardened
+AMIs - especially sor AMIs that conform to well-known security beckhmarks. See 
+[here](https://github.com/gravitational/teleport/issues/8078) for an example.
+
+## Details
+
+### Current Situation
+
+We  currently use [Packer](https://www.hashicorp.com/products/packer) and a shell
+script to provision our AMIs, based on Amazon Linux 2. The same Packerfile is used
+as a basis for several AMIs
+
+Inclusions for all AMIs:
+  * [Telegraf](https://www.influxdata.com/time-series-platform/telegraf/)
+    for capturing monitoring data
+  * [InfluxDB](https://www.influxdata.com/) as a data store for Telegraf
+  * [Grafana](https://grafana.com/) to display time series data
+  * [nginx](https://nginx.com) to serve Grafana's front end  
+  * EFF [`certbot`](https://https://certbot.eff.org/) for ACME Certificate
+    Rotation (used by optional ssm `systemd` units)
+  * Python 3 and `pip` for installing and running `certbot`
```

Comment:
Why is certbot installed via pip and not just baked into the AMI?

Reply:
IT is baked into the AMI, via pip.  I believe we use pip vs an RPM to support the route53 plugin, which (as of many years ago) was not packaged outside of the python ecosystem:

https://github.com/certbot/certbot/tree/3c667e8fffc2f6f4304f284ea52531bbf750a6c4/certbot-dns-route53

Reply:
It feels like we could copy this in from a builder rather than install it in the AMI via pip, but I did not review this in detail.  Let me know your thoughts

Reply:
I agree -- if we even need certbot, we can probably install it and its plugins via a prebuilt package.  I believe this is available in epel now.

Reply:
We **_do_** need `certbot` in the AMIs, as Teleport's ACME support can't handle sharing a single cert between multiple proxies and not everybody uses ACM.